src/components/permissions/hooks/usePermissions.js

```diff
@@ -0,0 +1,98 @@
+// @flow
+
+import { useEffect, useState } from 'react'
+
+import { noop } from 'lodash'
+
+import { type Permission, Permissions, PermissionStatuses } from '../types'
+
+import CameraPermissionsDialog from '../components/CameraPermissionsDialog'
+import ClipboardPermissionsDialog from '../components/ClipboardPermissionsDialog'
+
+import { useDialog } from '../../../lib/undux/utils/dialog'
+
+import api from '../api/PermissionsAPI'
+
+const usePermissions = (permission: Permission, callbacks = {}) => {
+  const [showDialog] = useDialog()
+  const { onAllowed = noop, onDenied = noop } = callbacks
```

Comment:
rename it to `options`, add  options to set custom dialogs (beforePromptPopup, permissionsDeniedPopup)